polybinder/simulate.py

```diff
@@ -364,178 +343,181 @@ def anneal(
+                    "If shrinking, all of  shrink_kT, shrink_steps and "
+                    "shrink_period need to be given."
+                )
+
+        device = hoomd.device.auto_select()
+        sim = hoomd.Simulation(device=device, seed=self.seed)
+        
+        if self.cg_system is False:
+            init_snap, forcefields, refs = create_hoomd_forcefield(
+                    structure=self.system,
+                    r_cut=self.r_cut,
+                    ref_distance=self.ref_distance,
+                    ref_mass=self.ref_mass,
+                    ref_energy=self.ref_energy,
+                    auto_scale=self.auto_scale,
             )
-        if shrink_steps is None:
-            shrink_steps = 0
+            if self.restart:
+                sim.create_state_from_gsd(self.restart)
+            else:
+                sim.create_state_from_snapshot(init_snap)
+        else:
+            init_snap, forcefields = self._create_hoomd_sim_from_snapshot()
+            sim.create_state_from_snapshot(init_snap)
 
         if not schedule:
             temps = np.linspace(kT_init, kT_final, len(step_sequence))
             temps = [np.round(t, 1) for t in temps]
             schedule = dict(zip(temps, step_sequence))
 
-        # Get hoomd stuff set:
-        hoomd_args = f"--single-mpi --mode={self.mode}"
-        sim = hoomd.context.initialize(hoomd_args)
-        with sim:
-            if self.cg_system is False:
-                objs, refs = create_hoomd_simulation(
-                    self.system,
-                    self.ref_distance,
-                    self.ref_mass,
-                    self.ref_energy,
-                    self.r_cut,
-                    self.auto_scale,
-                    nlist=self.nlist
-                )
-                init_x = objs[0].box.Lx
-                init_y = objs[0].box.Ly
-                init_z = objs[0].box.Lz
-            elif self.cg_system is True:
-                objs = self._create_hoomd_sim_from_snapshot()
-                self.log_quantities.remove("pair_lj_energy")
-                init_x = objs[0].configuration.box[0]
-                init_y = objs[0].configuration.box[1]
-                init_z = objs[0].configuration.box[2]
-
-            hoomd_system = objs[1]
-            init_snap = objs[0]
-            _all = hoomd.group.all()
-            hoomd.md.integrate.mode_standard(dt=self.dt)
-            hoomd.dump.gsd(
-                "sim_traj.gsd",
-                period=self.gsd_write,
-                group=_all,
-                phase=0,
-                dynamic=["momentum"],
-                overwrite=False,
+        init_x = init_snap.configuration.box[0]
+        init_y = init_snap.configuration.box[1]
+        init_z = init_snap.configuration.box[2]
+        if isinstance(self.nlist, hoomd.md.nlist.Tree):
+            forcefields[0].nlist = self.nlist(buffer=0.4)
+        forcefields[0].nlist.exclusions = ["bond", "1-3", "1-4"]
```

Comment:
You may not want this hardcoded. GAFF has 1-4 exclusions, but I'm not sure about other forcefields. `create_hoomd_forcefield` should do handle setting the exclusions automatically unless the 1-4 scaling factor is 0, and then it'll need to be set manually.

Reply:
Good point, do you know if I need to reset the exclusions when changing the nlist type, like changing it to a tree?

Reply:
I think so because you are creating a completely new object, so it would have no knowledge of the previous exclusions.